lib/declare_schema/model/field_spec.rb

```diff
@@ -112,14 +112,14 @@ def initialize(model, name, type, position: 0, **options)
 
         @options = Hash[@options.sort_by { |k, _v| OPTION_INDEXES[k] || 9999 }]
 
-        @sql_options = @options.except(*NON_SQL_OPTIONS)
+        @sql_options = @options.slice(*SQL_OPTIONS)
```

Comment:
I first tried `& SQL_OPTIONS` but that comes from Invoca::Utils. TIL that the ActiveSupport extension of `slice` to `Hash` was actually ported over to Ruby.